src/main/java/me/leoko/advancedban/manager/PunishmentManager.java

```diff
@@ -35,10 +35,22 @@ public void setup() {
             load(name, UUIDManager.get().getUUID(name), mi.getIP(player));
         }
     }
+    
+    public void refresh() {
+        MethodInterface mi = Universal.get().getMethods();
+        mi.runAsync(() -> {
+            punishments.clear();
+            history.clear();
+            for (Object player : mi.getOnlinePlayers()) {
+                String name = mi.getName(player).toLowerCase();
+                load(name, UUIDManager.get().getUUID(name), mi.getIP(player));
```

Comment:
the load-Method returns an IterimData-Object, you will need to call the .accept() method so that the IterimData merges with the punishments and history list